x-pack/plugins/lens/public/visualizations/datatable/visualization.tsx

```diff
@@ -537,6 +560,15 @@ export const getDatatableVisualization = ({
     };
   },
 
+  getTelemetryEventsOnSave(state, prevState) {
+    const colorMappingEvents = state.columns.flatMap((col) => {
+      const prevColumn = prevState?.columns?.find((prevCol) => prevCol.columnId === col.columnId);
+      return getColorMappingTelemetryEvents(col.colorMapping, prevColumn?.colorMapping);
+    });
+
+    return colorMappingEvents;
+  },
```

Comment:
@teresaalvarezsoler did you want to update these telemetry events to include the vis type they were used in? Currently all events are the same across `xy`, `partition`, `tagcloud` and now `datatable`. Example of the current events are...

- `lens_color_mapping_palette_eui_amsterdam_color_blind`
- `lens_color_mapping_unassigned_terms_neutral`
- `lens_color_mapping_colors_up_to_2`
- `lens_color_mapping_avg_count_terms_per_color_1`